src/Okta.Sdk.IntegrationTests/UserTypesClientShould.cs

```diff
@@ -0,0 +1,269 @@
+                replacedUserType.Name.Should().Be(testName);
+
+                var retrievedForValidation = await testClient.UserTypes.GetUserTypeAsync(createdUserType.Id);
+                retrievedForValidation.Id.Should().Be(createdUserType.Id);
+                retrievedForValidation.Description.Should().Be(replacedDescription);
+                retrievedForValidation.DisplayName.Should().Be(replacedDisplayName);
+            }
+            finally
+            {
+                await DeleteAllUserTypes();
+            }
+        }
+
+
+        [Fact]
+        public async Task DeleteUserTypeById()
+        {
+            var testClient = TestClient.Create();
+            var testDescription = $"{SdkPrefix}:{nameof(DeleteUserTypeById)} Test Description";
+            var testDisplayName = $"{SdkPrefix}:{nameof(DeleteUserTypeById)} Test DisplayName";
+            var testName = $"{SdkPrefix}_{nameof(DeleteUserTypeById)}_TestUserType";
+
+            var createdUserType = await testClient.UserTypes.CreateUserTypeAsync(new UserType
+            {
+                Description = testDescription,
+                DisplayName = testDisplayName,
+                Name = testName,
+            });
+
+            try
+            {
+                createdUserType.Id.Should().NotBeNullOrEmpty();
+
+                var retrievedUserType = await testClient.UserTypes.GetUserTypeAsync(createdUserType.Id);
+                retrievedUserType.Id.Should().Be(createdUserType.Id);
+
+                await testClient.UserTypes.DeleteUserTypeAsync(createdUserType.Id);
+
+                var ex = await Assert.ThrowsAsync<OktaApiException>(() =>
+                    testClient.UserTypes.GetUserTypeAsync(createdUserType.Id));
+                ex.StatusCode.Should().Be(404);
+            }
+            finally
+            {
+                await DeleteAllUserTypes();
+            }
+        }
+
+        [Fact]
+        public async Task ListAllUserTypes()
+        {
+            var testClient = TestClient.Create();
+            var existingUserTypeIds = new HashSet<string>();
+            await foreach (IUserType existingUserType in testClient.UserTypes.ListUserTypes())
+            {
+                existingUserTypeIds.Add(existingUserType.Id);
+            }
+
+            var testUserTypeIds = new HashSet<string>();
+            for (int i = 0; i < 5; i++)
+            {
+                var createdUserType = await testClient.UserTypes.CreateUserTypeAsync(new UserType()
+                {
+                    Description = $"{nameof(ListAllUserTypes)} Test Description ({i})",
+                    DisplayName = $"{nameof(ListAllUserTypes)} Test DisplayName ({i})",
+                    Name = $"{nameof(ListAllUserTypes)}_TestUserType_{i}",
+                });
+                testUserTypeIds.Add(createdUserType.Id);
+            }
+
+            try
+            {
+                HashSet<string> allUserTypeIds = new HashSet<string>();
```

Comment:
nit: In this method we have vars and explicitly typed vars, let's try to use the same coding rules when it's possible for consistency. In general, this SDK is using vars, so I would stick to it.